app/controllers/api_keys_controller.rb

```diff
@@ -0,0 +1,50 @@
+class ApiKeysController < ApplicationController
+  before_action :redirect_to_root, unless: :signed_in?
+  before_action :verify_authorization, only: :destroy
+
+  def index
+    @api_keys = current_user.api_keys
+    redirect_to new_profile_api_key_path if @api_keys.empty?
+  end
+
+  def new
+    @api_key = current_user.api_keys.build
+  end
+
+  def create
+    @key = SecureRandom.hex(16)
+    @api_key = current_user.api_keys.build(api_key_params)
+
+    if @api_key.save
+      flash[:notice] = t(".save_key", key: @key)
```

Comment:
What do you think about sending an email here to confirm the user has generated a new api key? It could help detect compromised accounts earlier.

Reply:
Until now we have only used email only for flows where email is a must (account verification or password reset).
We can take up precaustionary notification like these when we have email fail safe working.

Reply:
I think with latest changes (sending email on gem push for example) it will make sense to send an email here as well.